pkg/provisioningv2/rke2/planner/store.go

```diff
@@ -369,13 +391,15 @@ func (p *PlanStore) setMachineJoinURL(entry *planEntry, capiCluster *capi.Cluste
 }
 
 func getJoinURLFromOutput(entry *planEntry, capiCluster *capi.Cluster, rkeControlPlane *rkev1.RKEControlPlane) (string, error) {
-	if entry.Plan == nil || !IsEtcdOnlyInitNode(entry) || entry.Metadata.Annotations[rke2.JoinURLAnnotation] != "" {
```

Comment:
So we are saying that the joinURL of a node can change?

Reply:
Yes, it can change if IP addresses change and so forth.